### PR TITLE
refactor brew, steam, brewdetection and optocoupler

### PR DESCRIPTION
**Pull request by LoQue90**
this PR will remove the software brewdetection.
also steam over optocoupler for quickmill.
on default there will be no brewdetection anymore.
Since we change the behavior of the PID in total and we do not recommend heating during brew, there is no need for SW brewdetection.
On top there is a lot of code, that specialy is waiting because of SW brewdetection.
If someone want BREW PID, having some sort of hardware brewdetection (switch or optocoupler), is needed.
If we add brew detection with a reed sensor on the valve, the impact of adding a hardware brew detection is reduced.

During this PR i also want to rename Brewdection flags and variables, the fit our naming sheme and get this all a bit nicer.
Also there is maybe a bug that our main state mashine dosen´t travel back from ksteam to kpidnormal, when no brewdetection is enabled. This has to be fixed with this pr, or it isn´t possible to remove sw bredetection.

TODO:
Check code in total if anywhere some sw brewdecetion code is left
refactor brew, steam, brewdetection where neede, part of shotimer
rename variables
real world testing.

ANY help is highly welcome, feel free to comment and discuss. Or make PRs on my branch.

### CODE REVIEW

File: src/storage.h [hunk 1]
```diff
@@ -93,18 +93,18 @@ typedef struct __attribute__((packed)) {
     uint8_t freeToUse8[2];
     double pidTvBd;
     uint8_t freeToUse9[2];
-    double brewSwTimeSec;
+    double freeToUse10;
```
Comment (kjyv) [inline src/storage.h:96]: You can also rename some of previous ones too to get continuous counting

File: src/steamHandler.h [hunk 2]
```diff
@@ -20,20 +20,6 @@ void checkSteamSwitch() {
             if (steamSwitchReading == LOW && !steamFirstON) {
                 steamON = 0;
             }
-
```
Comment (kjyv) [inline src/steamHandler.h:23]: Is this really related to brew detection? It looks like this is steam mode handling for quickmill machines

File: src/main.cpp [hunk 3]
```diff
@@ -705,131 +616,6 @@ char *number2string(unsigned int in) {
     return number2string_uint;
 }
 
-
```
Comment (kjyv) [inline src/main.cpp:708]: this handles all brew detection, not just sw bd

File: src/main.cpp [hunk 4]
```diff
@@ -853,32 +639,6 @@ void setEmergencyStopTemp() {
     }
 }
 
-void initSteamQM() {
```
Comment (kjyv) [inline src/main.cpp:856]: This is also steam mode for QM, not brew detection and should probably stay

File: src/main.cpp [hunk 5]
```diff
@@ -1821,7 +1510,7 @@ void setup() {
         .type = kDouble,
         .section = sBDSection,
         .position = 20,
-        .show = [] { return true && FEATURE_BREWDETECTION == 1 && useBDPID; },
```
Comment (kjyv) [inline src/main.cpp:1824]: You'll probably want to remove all of these vars as their show condition is only true in case the BD feature is enabled and the flag useBDPID is true. That will never be the case when you remove the feature

File: src/main.cpp [hunk 6]
```diff
@@ -1136,52 +883,11 @@ void handleMachineState() {
...
         case kShotTimerAfterBrew:
-            brewDetection();
 
             if (millis() - lastBrewTimeMillis > SHOTTIMERDISPLAYDELAY) {
                 LOGF(INFO, "Shot time: %4.1f s", lastBrewTime / 1000);
-                machineState = kBrewDetectionTrailing;
-                lastBrewTime = 0;
-            }
-
-            if (steamON == 1) {
```
Comment (kjyv) [inline src/main.cpp:1147]: Why remove all of this?

File: src/embeddedWebserver.h [hunk 7]
```diff
@@ -60,7 +60,7 @@ void serverSetup();
 void setEepromWriteFcn(int (*fcnPtr)(void));
 
 // editable vars are specified in main.cpp
-#define EDITABLE_VARS_LEN 33
+#define EDITABLE_VARS_LEN 32
```
Comment (kjyv) [inline src/embeddedWebserver.h:63]: Shouldn't this be 31 (two vars removed)?

**Comment by kjyv** [comment]
It seems you're removing brew detection completely and also steam mode for quick mill - also the hardware based. Did I miss something or is this likely not what we want?

**Comment by LoQue90** [comment]
@kjyv didn't had time to work on this, so right now it's a mess.

We did discussed the quickmill steam stuff, because it was somehow not working anymore. In the end we did find a bit confusing how this works. One idea for quickmill thermoblock was to remove main power from steam switch and connect it to the esp, but also keep the pulsor board ad trigger that with a ssr directly from the steam switch.
That's why I removed also all the quickmill steam stuff.
But I'm total open for any other solutions.
After removing the quickmill stuff from the optocoupler, nothing realy was left from brewdetection. So I started to removed this in total.
My idea is: with Feature_Brewswitch we start reading the switch, with Brewcontrol_type 0 we only bring the state mashine into kbrew, start the shottimer and trigger Brew pid if needed.
If Brewcontrol_type is > 0 we add brew() or brewscale () to control brewing.
I put that idea on discord, but didn't get a lot responses to this, I'm open for input!
Gonna rename this into refactor for brewdetection, brewswitch, steamswitch, optocoupler aaaaaannnnd more :D
